integrations/spring/opendal-spring-boot-starter/src/main/java/org/apache/opendal/spring/config/OpenDALAutoConfiguration.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.opendal.spring.config;
+
+import org.apache.opendal.spring.core.DefaultOpenDAL;
+import org.apache.opendal.spring.core.OpenDAL;
+import org.apache.opendal.spring.core.OpenDALProperties;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+
+@AutoConfiguration
+@EnableConfigurationProperties(OpenDALProperties.class)
+public class OpenDALAutoConfiguration {
+    private final OpenDALProperties openDALProperties;
+
+    public OpenDALAutoConfiguration(OpenDALProperties openDALProperties) {
+        this.openDALProperties = openDALProperties;
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(OpenDAL.class)
+    public DefaultOpenDAL openGeminiTemplate(OpenDALProperties openDALProperties) {
```

Comment:
Is `Gemini` here a typo?

Reply:
@Xuanwo Yes, I am also developing another spring starter, fixed